zap/src/main/java/org/zaproxy/zap/eventBus/SimpleEventBus.java

```diff
@@ -246,6 +246,20 @@ public void publishSyncEvent(EventPublisher publisher, Event event) {
                         });
     }
 
+    @Override
+    public Set<String> getPublisherNames() {
+        return this.nameToPublisher.keySet();
```

Comment:
Wrap in unmodifiable set (same for following method), to prevent callers from changing it.